vertx-opentracing/src/main/java/io/vertx/tracing/opentracing/OpenTracingTracer.java

```diff
@@ -56,28 +58,37 @@ public OpenTracingTracer(boolean closeTracer, Tracer tracer) {
   }
 
   @Override
-  public <R> Span receiveRequest(Context context, R request, String operation,
-    Iterable<Map.Entry<String, String>> headers, TagExtractor<R> tagExtractor) {
-    SpanContext sc = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMap() {
-      @Override
-      public Iterator<Map.Entry<String, String>> iterator() {
-        return headers.iterator();
-      }
+  public <R> Span receiveRequest(Context context,
+                                 TracingPolicy policy,
+                                 R request,
+                                 String operation,
+                                 Iterable<Map.Entry<String, String>> headers,
+                                 TagExtractor<R> tagExtractor) {
+    if (policy != TracingPolicy.IGNORE) {
+      SpanContext sc = tracer.extract(Format.Builtin.HTTP_HEADERS, new TextMap() {
+        @Override
+        public Iterator<Map.Entry<String, String>> iterator() {
+          return headers.iterator();
+        }
 
-      @Override
-      public void put(String key, String value) {
-        throw new UnsupportedOperationException();
+        @Override
+        public void put(String key, String value) {
+          throw new UnsupportedOperationException();
+        }
+      });
+      if (sc != null || policy == TracingPolicy.SUPPORT) {
```

Comment:
"SUPPORT" means "ALWAYS" ? I don't really get why you chose that word, "always" seemed more clear to me

Reply:
it should be required here :-)

Reply:
but it will be always I guess